File: src/main/java/com/rarmash/cs2_simulator/skinspecs/Exterior.java
package com.rarmash.cs2_simulator.skinspecs;

public enum Exterior {
    FACTORY_NEW ("Factory New"),
    MINIMAL_WEAR ("Minimal Wear"),
    FIELD_TESTED ("Field-Tested"),
    WELL_WORN ("Well-Worn"),
    BATTLE_SCARRED ("Battle-Scarred");

    private String exteriorTitle;

    Exterior(String exteriorTitle) {
        this.exteriorTitle = exteriorTitle;
    }

    public String getExteriorTitle() {
        return exteriorTitle;
    }

    @Override
    public String toString() {
        return "Exterior {" +
                "exteriorTitle='" + exteriorTitle + '\'' +
                '}';
    }
}

File: src/main/kotlin/com/rarmash/cs2_simulator/model/enums/Exterior.kt
package com.rarmash.cs2_simulator.model.enums

enum class Exterior(
    val exteriorTitle: String,
    val min: Double,
    val max: Double
) {
    FACTORY_NEW("Factory New", 0.00, 0.07),
    MINIMAL_WEAR("Minimal Wear", 0.07, 0.15),
    FIELD_TESTED("Field-Tested", 0.15, 0.37),
    WELL_WORN("Well-Worn", 0.37, 0.44),
    BATTLE_SCARRED("Battle-Scarred", 0.44, 1.00);

    companion object {

        @JvmStatic
        fun fromFloat(value: Double): Exterior {
            return entries.firstOrNull { value in it.min..it.max }
                ?: throw IllegalArgumentException("Float value out of range: $value")
        }
    }

    override fun toString(): String = exteriorTitle
}